### PR TITLE
dev-dtvf-docs

### PR DESCRIPTION
**Pull request by mdhillman**
This PR brings in a major change to the documentation of the DTVF. Rather than using the GitHub wiki, documentation is now file-based and split across several files. Information has been updated, expanded upon (following recent feedback), and a new tutorial has been added.

In addition, DTVF images are now versioned correctly and include the compiled JS and CSS files (rather than relying on an external server to host them).

I believe the following needs testing/checking before the PR can be merged:

- Quick read through of the documentation (at least for errors, typos etc)
- Try running the example Mapbox and Cesium visualisations on their own
- Ensure that visualisations still function within a stack context

### CODE REVIEW

File: web/digital-twin-vis-framework/library/README.md [hunk 1]
```diff
@@ -1,58 +1,74 @@
...
+<br/>
+
+## Support
 
-- Run the `docker-compose -f docker/docker-compose.yml build deploy` command from within this directory.
-- The image should then be built and tagged as `ghcr.io/cambridge-cares/dtvf-base-image:latest`
-- Use the `docker push ghcr.io/cambridge-cares/dtvf-base-image:latest` command to upload the image.
-  - This requires having set up a GitHub token and storing it using the `docker login` command. 
\ No newline at end of file
+For support on developing the DTVF, please contact the support team at [CMCL](mailto:support@cmclinnovations.com).
```
Comment (sm453) [inline web/digital-twin-vis-framework/library/README.md:74]: Is the email address still current?

File: web/digital-twin-vis-framework/docs/tutorials.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* [Styling and Filtering 3D Tiles](https://cesium.com/learn/cesiumjs-learn/cesiumjs-3d-tiles-styling/)
+
+<br/>
+
+## DTVF tutorials
+
+These tutorials show how to gather, host, and display data using the DTVF. Note that these tutorials may have been written using older versions of the DTVF; however, the major functionality should still stand.
+
+* [Creating a Mapbox visualisation](./tutorial-mapbox.md)
+* [Creating a CesiumJS visualisation](./tutorial-cesium.md)
```
Comment (sm453) [inline web/digital-twin-vis-framework/docs/tutorials.md:37]: This link is broken. Please fix.

File: web/digital-twin-vis-framework/docs/tutorial-mapbox.md [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+## Gathering data
+
+In this case, we'll be grabbing the following data sets from the [NHS Website Datasets](https://www.nhs.uk/about-us/nhs-website-datasets/) and [NHS Digital](https://digital.nhs.uk/services/organisation-data-service/export-data-files/csv-downloads) sites.
+
+* `NHS Website Datasets` -> `GP and GP practice related data` -> `GP Practices`
+* `NHS Digital` -> `GP and GP practice related data` -> `GP Practitioners`
+  
+Download these and save/extract them on your local machine. The `GP Practices` data set should include a headered CSV file, and the `GP Practitioners` ZIP of an unheadered CSV and a PDF.
+
+If this data is no longer available on the NHS websites, please contact CMCL for an archived version.
```
Comment (sm453) [inline web/digital-twin-vis-framework/docs/tutorial-mapbox.md:26]: I'd remove this comment... (happy to discuss why)

File: web/digital-twin-vis-framework/docs/tutorial-mapbox.md [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+                }
+            ]
+        }
+    ]
+}
+```
+
+**3. Updating the settings file:**<br/>
+
+At the point, you may also want to update the visualisation's `setting.json` file to start the visualisation in a better location and at an appropriate zoom level to see all of England. Copy and paste the below into the `settings.json` file before refreshing the visualisation.
```
Comment (sm453) [inline web/digital-twin-vis-framework/docs/tutorial-mapbox.md:321]: "At this point"?

File: web/digital-twin-vis-framework/docs/tutorial-mapbox.md [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+
+**7. Updating the default content:**<br/>
+
+We can also update the default content in the side panel to better explain the purpose of our new visualisation. This can be done by updating the sections marked as `CUSTOMISABLE` in our visualisation's `index.html` file.
+
+Feel free to update the side panel content, saving the `index.html` file, then refreshing the visualisation to see the changes.
+
+**8. Reviewing the visualisation:**<br/>
+
+At the point, we've shown how to acquire, clean, map, upload, then visualise a new data set using The Stack and the DTVF. This constitutes the envisioned, basic use of the TWA tools to display data. By reading more of the available TWA documentation, users can learn how to add more data sets, produce more complex styling, and build more interactive visualisations.
```
Comment (sm453) [inline web/digital-twin-vis-framework/docs/tutorial-mapbox.md:413]: "At this point"?

File: web/digital-twin-vis-framework/example-mapbox-vis/README.md [hunk 6]
```diff
@@ -114,23 +19,57 @@ A small amount of sample data has been committed to demonstrate the power of the
...
+In most deployed visualisations, an online stack of microservices will provide data endpoints through which data can be queried/loaded onto the visualisation. In this example, no online stack is used, solely to remove a lengthy prerequisite step. Instead, sample data in local GeoJSON files have been added (to be hosted by the visualisation's web server) and, in one case, a community provided WMS endpoint connected to.
+
+In production, it is advised that all data is loaded into a geospatial data provider (like GeoServer) and a WMS endpoint used; local files can be utilised but then do not offer the optimisation and caching of services like GeoServer. For more information on how to do this, see the README for the [Stack Data Uploader](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-data-uploader).
+
+It's also worth noting that with this example visualisation, no stack or FeatureInfoAgent is running, hence these is no support for dynamic metadata or timeseries is available. This is something that we plan to work on in future.
```
Comment (sm453) [inline web/digital-twin-vis-framework/example-mapbox-vis/README.md:30]: "hence there is" ... "timeseries available"

File: web/digital-twin-vis-framework/example-cesium-vis/README.md [hunk 7]
```diff
@@ -167,27 +13,54 @@ A small amount of sample data has been committed to demonstrate the power of the
 - **New York**:
   - Tiled 3D buildings, loaded from a remote CMCL server.
   - 2D river data from a WMS endpoint provided by Cornell University.
+  - Sample tile sets to demonstrate clipping planes.
   - No metadata or timeseries present in this data set.
-  - Styling based on distance from Castle Clinton
-  - Filtering to hide the Whitehall Ferry terminal
 
-It's worth noting that with this sample data, no stack is running so no support for dynamic metadata or timeseries is available. This is something that we plan to work on in future, no true generic solution exists for this so far.
+It's also worth noting that with this example visualisation, no stack or FeatureInfoAgent is running, hence these is no support for dynamic metadata or timeseries is available. This is something that we plan to work on in future.
```
Comment (sm453) [inline web/digital-twin-vis-framework/example-cesium-vis/README.md:19]: "hence there is" ... "timeseries available"

**Review by sm453 — CHANGES_REQUESTED** [review]

File: web/digital-twin-vis-framework/docs/img/filtering-example.png
[the remark this answers is not in the document]
Comment (mdhillman) [reply web/digital-twin-vis-framework/docs/img/filtering-example.png:1]: Replaced with smaller JPG.

File: web/digital-twin-vis-framework/docs/advanced-cesium.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+- If adding planes to an existing visualisation, ensure the version of Cesium JS and CSS files used in the `index.html` file are _at least_ 1.105.
+
+<br/>
+<p align="center">
+    <img src="./img/sample-nyc-3.JPG" alt="Example of a clipping plane on a 3D tileset" width="75%"/>
+</p>
+
+<br/>
+
+## Data driven styling
```
[the remark this answers is not in the document]
Comment (mdhillman) [reply web/digital-twin-vis-framework/docs/advanced-cesium.md:56]: Added missing section.

File: web/digital-twin-vis-framework/docs/cesium.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+* Free, client-side, JavaScript library.
+* Does not require any API key or authentication.
+* Provides basic functionality to display and style 3D data.
+* Provides no map imagery or terrain elevation data.
+
+**Cesium ion:**
+* Not used by the DTVF or the wider TWA.
+* Extension to the CesiumJS library.
+* Premium offering, requires payment for commercial or funded-education use.
+* Required an API key.
```
[the remark this answers is not in the document]
Comment (mdhillman) [reply web/digital-twin-vis-framework/docs/cesium.md:26]: Updated.

File: web/digital-twin-vis-framework/docs/cesium.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
+# Working with CesiumJS
+
+The DTVF supports a number of different data formats and representations; in the case of 3D data, a range of features offered by [CesiumJS](https://cesium.com/platform/cesiumjs/) are supported through the DTVF. 
+
+Unlike the Mapbox library, the CesiumJS library does not support specification of data in JSON format. Data and styling options must be loaded by direct method calls. As such, not every piece of functionality in CesiumJS has been integrated so that it can be used via the DTVF's configuration files. Basic visualisation of a number of data formats has been enabled, and requests can always be made to add support for other features.
+
+CesiumJS is the primary visualisation provider for 3D building data within The World Avatar. It has a number of useful features, but can be difficult to get to groups with. If possible, it's recommended that users create DTVF visualisations using Mapbox, rather than CesiumJS, unless display of 3D building data is required.
```
[the remark this answers is not in the document]
Comment (mdhillman) [reply web/digital-twin-vis-framework/docs/cesium.md:7]: Updated.

File: web/digital-twin-vis-framework/library/README.md (hunk 1 above)
Comment (mdhillman) [reply web/digital-twin-vis-framework/library/README.md:74]: Updated.

File: web/digital-twin-vis-framework/docs/tutorial-mapbox.md (hunk 4 above)
Comment (mdhillman) [reply web/digital-twin-vis-framework/docs/tutorial-mapbox.md:321]: Updated.

File: web/digital-twin-vis-framework/docs/tutorial-mapbox.md (hunk 5 above)
Comment (mdhillman) [reply web/digital-twin-vis-framework/docs/tutorial-mapbox.md:413]: Updated.

File: web/digital-twin-vis-framework/example-mapbox-vis/README.md (hunk 6 above)
Comment (mdhillman) [reply web/digital-twin-vis-framework/example-mapbox-vis/README.md:30]: Updated.

File: web/digital-twin-vis-framework/example-cesium-vis/README.md (hunk 7 above)
Comment (mdhillman) [reply web/digital-twin-vis-framework/example-cesium-vis/README.md:19]: Updated.

File: web/digital-twin-vis-framework/library/VERSION [hunk 11]
```diff
@@ -0,0 +1 @@
+3.6.0
```
[the remark this answers is not in the document]
Comment (mdhillman) [reply web/digital-twin-vis-framework/library/VERSION]: Now updated to 3.8.0

**Comment by cmclscourtney** [comment]
In future, it would be great if versions of the DTVF on branches followed the ```1.0.1-123-SNAPSHOT```. 
Also the stack tools currently uses the ```latest``` tag as the built-in config. The version on main should always be pushed as both the latest tag and the version number.

**Comment by mdhillman** [comment]
> In future, it would be great if versions of the DTVF on branches followed the `1.0.1-123-SNAPSHOT`. Also the stack tools currently uses the `latest` tag as the built-in config. The version on main should always be pushed as both the latest tag and the version number.

Agreed (although I don't tend to push the Docker image until it's merged to master).

I've update the GitHub action that automatically uploads the image after a merge to master to now push with two tags: one with the explicit version number, the other "latest".

**Review by cmclscourtney — APPROVED** [review]

**Comment by mdhillman** [comment]
Feedback from @sm453 has been implemented, technical changes have been tested & approved by @cmclscourtney.

**Pull request merged**